packages/preferences/src/browser/preferences-editor.tsx

```diff
@@ -0,0 +1,567 @@
+     * such as defaultValue, name, category, description and enum
+     * and then set them to our defaultPreferences
+     * Then we get the current value from the [PreferenceService] and set it at the current value
+     * and then return a combined object with `id, name, category` and data being the remaining
+     * NOTE: We can't specify the scope of the where we get the current values from so how do we
+     * handle what value we show if we dont know where they from? as a value can exist on multiple scopes
+     */
+
+    getPreference(id: string): { value: any, [key: string]: any } {
+        const data = this.properties[id];
+        const value = this.getPreferenceValue(id);
+        return { value, ...data };
+    }
+
+    getPreferenceValue(id: string): any {
+        const values = this.service.inspect(id) as { [prop: string]: any };
+        if (values) {
+            const rawKey = `${PreferenceScope[this.scopes[this.activeScope]].toLowerCase()}Value`;
+            const key = rawKey.replace('user', 'global');
+            return values[key] !== undefined ? values[key] : values.defaultValue;
+        }
+        return undefined;
+    }
+
+    protected handleSetPreference = (e: React.ChangeEvent | React.FormEvent): void => {
+        const target = (e.target as HTMLInputElement);
+        const id = target.attributes.getNamedItem('data-preference-id');
+        if (id) {
+            const { type } = this.getPreference(id.value);
+            const value = target.type !== 'checkbox' ?
+                this.castStringToType(type, target.value) :
+                target.checked;
+            this.setPreference(id.value, value);
+        }
+    };
+
+    protected setPreferenceDebounced = debounce((id: string, value: any): Promise<void> => this.setPreference(id, value), 500);
+
+    protected setPreference = async (id: string, value: any): Promise<void> => {
+        const scope: any = this.scopes[this.activeScope];
+        await this.service.set(id, value, scope);
+        this.messageService.log('Preferences saved', { timeout: 3000 });
+        this.update();
+    };
+
+    resetPreferences(): void {
+        this.properties = this.defaultPreferences;
+    }
+
+    protected handleChangeScope = ({ target: { value } }: React.ChangeEvent<HTMLSelectElement>): void => {
+        // This is here to avoid using the synthetic event asynchronously
+        this.changeScope(value);
+    };
+
+    public openJSONEditor = (): void => {
+        switch (this.activeScope) {
+            case 0:
+                this.editorManager.open(USER_PREFERENCE_URI);
+                break;
+            case 1:
+                const workspace = this.workspaceService.workspace;
+                const workspaceRootUri = this.workspaceService.getWorkspaceRootUri(workspace && new URI(workspace.uri));
+                console.log('workspace:', workspaceRootUri); // eslint-disable-line no-console
```

Comment:
leftovers?  please check that you don't have such

Reply:
Will do. Sorry about that.